matrix_benchmarking/regression.py

```diff
@@ -0,0 +1,176 @@
+
+        if not self.new_payload:
+            return [{"result": None, "kpi": None, "regression": vars(RegressionStatus(0, explanation="Not enough new data"))}]
+
+        if not self.lts_payloads:
+            return [{"result": None, "kpi": None, "regression": vars(RegressionStatus(0, explanation="Not enough LTS data"))}]
+
+        regression_results = []
+
+        kpis_to_test = vars(self.new_payload.results.lts.kpis).keys() if not self.kpis else self.kpis
+        for kpi in kpis_to_test:
+
+            curr_values = vars(self.new_payload.results.lts.kpis)[kpi].value
+            lts_values = list(map(lambda x: vars(x.results.kpis)[kpi].value, self.lts_payloads))
+
+            if type(vars(self.new_payload.results.lts.kpis)[kpi].value) is list:
+                if kpi in self.combine_funcs:
+                    curr_values = self.combine_funcs[kpi](curr_values)
+                    lts_values = [self.combine_funcs[kpi](v) for v in lts_values]
+                else:
+                    logging.warning(f"Skipping KPI with list of values, consider filtering KPIs or providing a combine_func for {kpi}")
+                    continue
+
+            regression_results.append(
+                {
+                    "result": self.new_payload.get_settings(),
+                    "kpi": kpi,
+                    "regression": vars(
+                        self.regression_test(
+                            curr_values,
+                            lts_values
+                        )
+                    )
+                }
+            )
+        return regression_results
+
+    def regression_test(self, new_result: float, lts_result: np.array) -> RegressionStatus:
+        return RegressionStatus(0, explanation="Default return status")
+
+
+class ZScoreIndicator(RegressionIndicator):
+    """
+    Example regression indicator that uses the Z score as a metric
+    to determine if the recent test was an outlier
+    """
+    def __init__(self, *args, threshold=3, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.threshold = threshold
+
+    def regression_test(self, new_result: float, lts_results: np.array) -> RegressionStatus:
+        """
+        Determine if the curr_result is more/less than threshold
+        standard deviations away from the previous_results
+        """
+        mean = np.mean(lts_results)
+        std = np.std(lts_results)
+        z_score = (new_result - mean) / std
+        if abs(z_score) > self.threshold:
+            return RegressionStatus(
+                1,
+                direction=1 if z_score > 0 else -1,
+                explanation="z-score greater than threshold",
+                details={"threshold": self.threshold, "zscore": z_score}
+            )
+        else:
+            return RegressionStatus(
+                0,
+                explanation="z-score not greater than threshold",
+                details={"threshold": self.threshold, "zscore": z_score}
+            )
```

Comment:
can you try to have a single return here, and use variables to save the explanation/status
that will make the code more readable (=always return a `RegressionStatus`, but if above threshold, set a different status/explanation)